src/lsm/table_memory.zig

```diff
@@ -0,0 +1,233 @@
+const std = @import("std");
+const mem = std.mem;
+const math = std.math;
+const assert = std.debug.assert;
+
+const constants = @import("../constants.zig");
+const binary_search = @import("binary_search.zig");
+
+pub fn TableMemoryType(comptime Table: type) type {
+    const Key = Table.Key;
+    const Value = Table.Value;
+    const value_count_max = Table.value_count_max;
+    const compare_keys = Table.compare_keys;
+    const key_from_value = Table.key_from_value;
+
+    return struct {
+        const TableMemory = @This();
+
+        pub const ValueContext = struct {
```

Comment:
What about `TableMemoryScope` as a name, to make it more obvious what this is for?

Reply:
I'm not sure - to me that implies it's used only if a scope is active or it has some specific relation to scope support.